snuba/web/split.py

```diff
@@ -1,10 +1,16 @@
 import copy
```

Comment:
There is no reason for this file to be in /web. I left it there only to make this PR easier to review since github shows the change with respect to the previous split logic instead of showing a new file. (I tried to move the file first but somehow after a few commits, github lost track the file was moved and just showed a brand new file).
Also I kept the refactoring of this class to the very minimum needed to move the splitting logic to make the PR easier to review. I think more refactoring is worth doing.

Moving the file to where it should be will be trivial

Reply:
Let's keep that as a separate change, I don't want to derail another otherwise good change with what does and doesn't belong in web again. 😞

Reply:
Unfortunately, if I move the file here (I tried), github will just show you a brand new file with no connection to the previous one (still don't get why, it should understand I did a git mv).
That would make it really hard to spot what actually changed in the split code which is already complex.

Reply:
https://github.com/getsentry/snuba/pull/883